File: Common/Sponsor.cs
using System.Runtime.Serialization;

namespace DevSpace.Common {
	[DataContract]
	[KnownType( typeof( SponsorLevel ) )]
	public class Sponsor : ISponsor {
		private Sponsor() {}

		[DataMember]public int Id { get; private set; }

		[DataMember( Name = "Level" )]private SponsorLevel _level;
		public ISponsorLevel Level { get { return _level; } }

		[DataMember]	public string DisplayName { get; private set; }
		[DataMember]public string LogoLarge { get; private set; }
		[DataMember]public string LogoSmall { get; private set; }
	}
}

File: FSharp.Common/Sponsor.fs
namespace DevSpace.FSharp.Common

type Sponsor =
  {
    Id:int;
    Level:SponsorLevel;
    DisplayName:string;
    LogoLarge:string;
    LogoSmall:string;
    Website:string
  }

  member this.WithId( value ) =
    { this with Id = value }

  member this.WithLevel( value ) =
    { this with Level = value }

  member this.WithDisplayName( value ) =
    { this with DisplayName = value }

  member this.WithLogoLarge( value ) =
    { this with LogoLarge = value }

  member this.WithLogoSmall( value ) =
    { this with LogoSmall = value }

  member this.WithWebsite( value ) =
    { this with Website = value }
